python/cudf/cudf/tests/test_numpy_interop.py

```diff
@@ -71,3 +71,37 @@ def test_numpy_non_contiguious():
 
     gdf = DataFrame.from_records(rec, index="index")
     assert_eq(aa, gdf["a"].values)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        Series([1, 2, 3, -12, 12, 44]),
+        Series([1, 2, 3, -12, 12, 44], dtype="str"),
+        Series([1, 2, 3, -12, 12, 44])._column,
+        Series([1, 2, 3, -12, 12, 44], dtype="str")._column,
+        Series([1, 2, 3, -12, 12, 44]).index,
+        DataFrame({"a": [1, 2, 3, -1234], "b": [0.1, 0.2222, 0.4, -3.14]}),
+        DataFrame(
+            {"a": [1, 2, 3, -1234], "b": [0.1, 0.2222, 0.4, -3.14]}
+        ).index,
+    ],
+)
+@pytest.mark.parametrize("dtype", [None, "float", "int", "str"])
+def test_series_dataframe__array__(data, dtype):
+    gs = data
+    ps = data.to_pandas()
+
+    if isinstance(gs, DataFrame):
+        assert_eq(
+            ps.__array__(dtype=dtype).astype(dtype), gs.__array__(dtype=dtype)
+        )
+        assert (
```

Comment:
Doesn't `assert_eq` already check for the equality of dtypes?

Reply:
Actually yes, removed it. 👍